src/main/java/cc/protea/util/http/Utf8Request.java

```diff
@@ -0,0 +1,305 @@
+    }
+
+    /**
+     * Removes the specified Query Parameter.
+     *
+     * @param name
+     *            The name of the Query Parameter to remove
+     * @return this Request, to support chained method calls
+     */
+    public Utf8Request removeQueryParameter(final String name) {
+        this.query.remove(name);
+        return this;
+    }
+
+    /**
+     * Issues a GET to the server.
+     *
+     * @return The {@link Response} from the server
+     * @throws IOException
+     */
+    public Response getResource() throws IOException {
+        buildQueryString();
+        buildHeaders();
+
+        connection.setDoOutput(true);
+        connection.setRequestMethod("GET");
+
+        return readResponse();
+    }
+
+    /**
+     * Issues a PUT to the server.
+     *
+     * @return The {@link Response} from the server
+     * @throws IOException
+     */
+    public Response putResource() throws IOException {
+        return writeResource("PUT", this.body);
+    }
+
+    public Response headResource() throws IOException {
+        buildQueryString();
+        buildHeaders();
+
+        connection.setDoOutput(true);
+        connection.setRequestMethod("HEAD");
+
+        return readResponse();
+    }
+
+    public Response optionsResource() throws IOException {
+        buildQueryString();
+        buildHeaders();
+
+        connection.setDoOutput(true);
+        connection.setRequestMethod("OPTIONS");
+
+        return readResponse();
+    }
+
+    public Response traceResource() throws IOException {
+        buildQueryString();
+        buildHeaders();
+
+        connection.setDoOutput(true);
+        connection.setRequestMethod("TRACE");
+
+        return readResponse();
+    }
+
+    /**
+     * Issues a POST to the server.
+     *
+     * @return The {@link Response} from the server
+     * @throws IOException
+     */
+    public Response postResource() throws IOException {
+        return writeResource("POST", this.body);
+    }
+
+    /**
+     * Issues a DELETE to the server.
+     *
+     * @return The {@link Response} from the server
+     * @throws IOException
+     */
+    public Response deleteResource() throws IOException {
+        buildQueryString();
+        buildHeaders();
+
+        connection.setDoOutput(true);
+        connection.setRequestMethod("DELETE");
+
+        return readResponse();
+    }
+
+    /**
+     * A private method that handles issuing POST and PUT requests
+     *
+     * @param method
+     *            POST or PUT
+     * @param body
+     *            The body of the Message
+     * @return the {@link Response} from the server
+     * @throws IOException
+     */
+    private Response writeResource(final String method, final String body) throws IOException {
+        buildQueryString();
+        buildHeaders();
+
+        connection.setDoOutput(true);
+        connection.setRequestMethod(method);
+
+        writer = new OutputStreamWriter(connection.getOutputStream(), "UTF-8");
```

Comment:
This is the significant change. It calls the constructor with two parameters - the second being the character set to support, overriding the system default (which shouldn't be used in an SDK).